src/App.tsx

```diff
@@ -1,61 +1,153 @@
+import { useState } from 'react';
+
+import { TodoList } from './components/TodoList';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (userId: number): User | null => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const newTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+let maxId = Math.max(...newTodos.map(({ id }) => id));
 
 export const App = () => {
+  const [todos, setTodos] = useState(newTodos);
+  const [user, setUser] = useState('0');
+  const [title, setTitle] = useState('');
+
+  const [isValidUser, setIsValidUser] = useState(true);
+  const [isValidTitle, setIsValidTitle] = useState(true);
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUser(value);
+    setIsValidUser(true);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+    setIsValidTitle(true);
+  };
+
+  const resetForm = () => {
+    setTitle('');
+    setUser('0');
+  };
+
+  const handleSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let isValidValues = true;
+
+    const titleOnlyWithLetters = title.replace(/[0-9]/g, '')
+      .split(' ')
+      .filter(letter => letter !== '')
+      .join(' ');
```

Comment:
You can use [string.trim()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim)